esmtools/prediction.py

```diff
@@ -609,9 +604,78 @@ def PM_compute(ds, control, metric=_pearson_r, comparison=_m2m, anomaly=False,
         raise ValueError('specify metric argument')
 
 
+def DP_compute_skill(dp, reference, nlags=None):
+    """
+    Compute a predictability skill score for a reconstruction framework dataset.
+    
+    This is currently set up to just run a simple correlation (i.e., ACC) but
+    other metrics should be implemented if needed in the future (as in PM_compute)
+
+    Note that if reference is the reconstruction, the output correlation coefficients
+    are for potential predictability. If the reference is observations, the ouput
+    correlation coefficients are actual skill.
+
+    Parameters
+    ----------
+    dp : xarray object
+        Expected to follow package conventions (and should be an ensemble mean)
+        `ensemble` : dim of initialization dates
+        `time` : dim of lead years from those initializations
+        Additional dims can be lat, lon, depth, etc. but should not be individual
+        members.
+    reference : xarray object
+        Reconstruction or observations over same time period
+    nlags : int (default length of `time` dim)
+        How many lags to compute skill/potential predictability out to
+
+    Returns
+    -------
+    skill : xarray object
+        Predictability with main dimension `lag`
+    """
+    _check_xarray(dp)
+    _check_xarray(reference)
+    if 'member' in _get_dims(dp):
+        print("Taking ensemble mean...")
+        dp = dp.mean('member')
+    N = dp['ensemble'].size
+    if nlags is None:
+        nlags = dp['time'].size
+    skill = xr.concat([xr_corr(dp.isel(time=i), reference, lag=i,
```

Comment:
it would be nice to have this generalized for more metrics, at least also _rmse which follows the same requirements as _pearson_r.

I have a function in mind where we can choose the metric for skill forecast. See my comment in #21 

Some thing like. and then we can replace _rmse or xr_autocorr with metrics.

```skill=xr.concat([_rmse(dp.isel(time=i),reference,'ensemble') for i in range(DPE.time.size)],'time')```
But in this line I forgot to shift the `ensemble` dimension of assim (I think thats needed for _pearson_r and _rmse to work. That feature is built-in `xr_autocorr` and `xr_corr` and I tried to get it out into `_shift`)
I think if you can somehow rewrite xr_autocorr in the same way as xskillscore.pearson_r or xs.rmse, then we are there.

Reply:
something like: this now even works for both comparison types

```python
def hindcast_compute(ds, reference, metric=_pearson_r):
    skilllist=[]
    for i in range(ds.time.size):
        a,b=_shift(ds.isel(time=i),assim,lag=i,dim='ensemble')
        if 'member' in ds:
            a,b=xr.broadcast(a,b)
            a=a.stack(svd=('ensemble','member'))
            b=b.stack(svd=('ensemble','member'))
            skilllist.append(metric(a,b,'svd'))
        else:
            skilllist.append(metric(a,b,'ensemble'))
    skill = xr.concat(skilllist,dim='time')
    return skill

hindcast_compute(DPE, assim)['SST'].plot()
hindcast_compute(DPE.mean('member'), assim)['SST'].plot()
```

Another future step (next PR) would be to take only a few selected ensembles (init years) and see if anomalous starting conditions influences the skill.

Reply:
Thanks for all these comments. I am going to go through them and try to incorporate what I can. At some point I will probably merge what is working since this pull request is getting huge. Then we can open a new one that focuses on the things I missed.

Reply:
That’s the m2a example here in if member in ds and the ensemble mean version without computing it below. We are very close to finishing this. The I will just go over the PM stuff later. But great work so far.

Reply:
I think with this we can merge then.